# Práctica 1 - Limpieza de Datos

<b>Equipo: </b> 1 <br> 
<b>Grupo: </b> 002 <br> 
<b>Integrantes: </b> 
- 1794656 Félix Barreras Jesús Guillermo
- 1723237 Linares Núñez Vanessa Stefanía
- 1749744 Ramirez Vargas Dayan
- 1803744 Rodríguez Estrada Erick Alejandro

---
## Base de datos 1

### Titulo de la base de datos.

<b> a) Nombre con el que se encuentra en la página encontrada: </b> World Happiness Report <br>
<b> b) Url de la página: </b> https://www.kaggle.com/unsdsn/world-happiness?select=2019.csv

### Preparación de datos
1. Importar librerías

In [1]:
import pandas as pd
import numpy as np
import json

2. Importar base de datos:

In [2]:
df = pd.read_csv("uncleandata_happiness_2019.csv")

>Recordar que para que el archivo csv pueda ser leído apropiadamente se debe de encontrar en la misma carpeta(direccion) que nuestro archivo de python

### Visualización de los Datos


In [3]:
df

Overall rank         Country or region  Score  GDP per capita  \
0               1                   Finland  7.769           1.340   
1               2                   Denmark  7.600           1.383   
2               3                    Norway  7.554           1.488   
3               4                   Iceland  7.494           1.380   
4               5               Netherlands  7.488           1.396   
..            ...                       ...    ...             ...   
151           152                    Rwanda  3.334           0.359   
152           153                  Tanzania  3.231           0.476   
153           154               Afghanistan  3.203           0.350   
154           155  Central African Republic  3.083           0.026   
155           156               South Sudan  2.853           0.306   

     Social support  Healthy life expectancy  Freedom to make life choices  \
0             1.587                    0.986                         0.596   
1             1.573                    0.996                         0.592   
2             1.582                    1.028                         0.603   
3             1.624                    1.026                         0.591   
4             1.522                    0.999                         0.557   
..              ...                      ...                           ...   
151           0.711                    0.614                         0.555   
152           0.885                    0.499                         0.417   
153           0.517                    0.361                         0.000   
154           0.000                    0.105                         0.225   
155           0.575                    0.295                         0.010   

     Generosity  Perceptions of corruption  
0         0.153                      0.393  
1         0.252                      0.410  
2         0.271                      0.341  
3         0.354                      0.118  
4         0.322                      0.298  
..          ...                        ...  
151       0.217                      0.411  
152       0.276                      0.147  
153       0.158                      0.025  
154       0.235                      0.035  
155       0.202                      0.091  

[156 rows x 9 columns]

1. Revisar cuales son los datos que conforman la base de datos:

In [4]:
df.shape
print("Numero de filas:", df.shape[0], "\nNumero de columnas:", df.shape[1])

Numero de filas: 156 
Numero de columnas: 9


2. Ver los primeros cinco elementos de cada una de las columnas

In [5]:
df.head()

Overall rank Country or region  Score  GDP per capita  Social support  \
0             1           Finland  7.769           1.340           1.587   
1             2           Denmark  7.600           1.383           1.573   
2             3            Norway  7.554           1.488           1.582   
3             4           Iceland  7.494           1.380           1.624   
4             5       Netherlands  7.488           1.396           1.522   

   Healthy life expectancy  Freedom to make life choices  Generosity  \
0                    0.986                         0.596       0.153   
1                    0.996                         0.592       0.252   
2                    1.028                         0.603       0.271   
3                    1.026                         0.591       0.354   
4                    0.999                         0.557       0.322   

   Perceptions of corruption  
0                      0.393  
1                      0.410  
2                      0.341  
3                      0.118  
4                      0.298

3. Ver los últimos cinco elementos de cada una de las columnas

In [6]:
df.tail()

Overall rank         Country or region  Score  GDP per capita  \
151           152                    Rwanda  3.334           0.359   
152           153                  Tanzania  3.231           0.476   
153           154               Afghanistan  3.203           0.350   
154           155  Central African Republic  3.083           0.026   
155           156               South Sudan  2.853           0.306   

     Social support  Healthy life expectancy  Freedom to make life choices  \
151           0.711                    0.614                         0.555   
152           0.885                    0.499                         0.417   
153           0.517                    0.361                         0.000   
154           0.000                    0.105                         0.225   
155           0.575                    0.295                         0.010   

     Generosity  Perceptions of corruption  
151       0.217                      0.411  
152       0.276                      0.147  
153       0.158                      0.025  
154       0.235                      0.035  
155       0.202                      0.091

4. Ver los nombres de las columnas de nuestra base de datos

In [7]:
df.columns

Index(['Overall rank', 'Country or region', 'Score', 'GDP per capita',
       'Social support', 'Healthy life expectancy',
       'Freedom to make life choices', 'Generosity',
       'Perceptions of corruption'],
      dtype='object')

5. Cambiar los nombres de las columnas a uno que nos facilite la extracción de su información

In [28]:
df = df.rename(columns={'Overall rank': 'Clasificacion_General', 
                              'Country or region':'Pais_o_region',
                              'Score':'Puntuacion',
                              'GDP per capita': 'PIB_per_capita',
                              'Social support':'Apoyo_Social',
                              'Healthy life expectancy':'Esperanza_de_vida', 
                              'Freedom to make life choices':'Libertad',
                              'Generosity':'Generocidad',
                              'Perceptions of corruption':'Percepción_de_corrupcion'})

df

Clasificacion_General             Pais_o_region  Puntuacion  \
0                        1                   Finland       7.769   
1                        2                   Denmark       7.600   
2                        3                    Norway       7.554   
3                        4                   Iceland       7.494   
4                        5               Netherlands       7.488   
..                     ...                       ...         ...   
151                    152                    Rwanda       3.334   
152                    153                  Tanzania       3.231   
153                    154               Afghanistan       3.203   
154                    155  Central African Republic       3.083   
155                    156               South Sudan       2.853   

     PIB_per_capita  Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0             1.340         1.587              0.986     0.596        0.153   
1             1.383         1.573              0.996     0.592        0.252   
2             1.488         1.582              1.028     0.603        0.271   
3             1.380         1.624              1.026     0.591        0.354   
4             1.396         1.522              0.999     0.557        0.322   
..              ...           ...                ...       ...          ...   
151           0.359         0.711              0.614     0.555        0.217   
152           0.476         0.885              0.499     0.417        0.276   
153           0.350         0.517              0.361     0.000        0.158   
154           0.026         0.000              0.105     0.225        0.235   
155           0.306         0.575              0.295     0.010        0.202   

     Percepción_de_corrupcion  
0                       0.393  
1                       0.410  
2                       0.341  
3                       0.118  
4                       0.298  
..                        ...  
151                     0.411  
152                     0.147  
153                     0.025  
154                     0.035  
155                     0.091  

[156 rows x 9 columns]

6. Conocer el tipo de dato que contiene cada columna

In [9]:
df.dtypes

Clasificacion_General         int64
Pais_o_region                object
Puntuacion                  float64
PIB_per_capita              float64
Apoyo_Social                float64
Esperanza_de_vida           float64
Libertad                    float64
Generocidad                 float64
Percepción_de_corrupcion    float64
dtype: object

- Si queremos conocer el tipo de datos de una columna en específico:

In [10]:
df['Puntuacion'].dtype

dtype('float64')

### Descripción Básica de las columnas

In [11]:
df

Clasificacion_General             Pais_o_region  Puntuacion  \
0                        1                   Finland       7.769   
1                        2                   Denmark       7.600   
2                        3                    Norway       7.554   
3                        4                   Iceland       7.494   
4                        5               Netherlands       7.488   
..                     ...                       ...         ...   
151                    152                    Rwanda       3.334   
152                    153                  Tanzania       3.231   
153                    154               Afghanistan       3.203   
154                    155  Central African Republic       3.083   
155                    156               South Sudan       2.853   

     PIB_per_capita  Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0             1.340         1.587              0.986     0.596        0.153   
1             1.383         1.573              0.996     0.592        0.252   
2             1.488         1.582              1.028     0.603        0.271   
3             1.380         1.624              1.026     0.591        0.354   
4             1.396         1.522              0.999     0.557        0.322   
..              ...           ...                ...       ...          ...   
151           0.359         0.711              0.614     0.555        0.217   
152           0.476         0.885              0.499     0.417        0.276   
153           0.350         0.517              0.361     0.000        0.158   
154           0.026         0.000              0.105     0.225        0.235   
155           0.306         0.575              0.295     0.010        0.202   

     Percepción_de_corrupcion  
0                       0.393  
1                       0.410  
2                       0.341  
3                       0.118  
4                       0.298  
..                        ...  
151                     0.411  
152                     0.147  
153                     0.025  
154                     0.035  
155                     0.091  

[156 rows x 9 columns]

<b>Descripción: </b> 

1. Clasificación General:

La columna de clasificación general indica el ranking de cada país, siendo el número 1 el más feliz de la muestra y el 156 el menos. Es de caracter númerico y va de 1 al 156 sin repetir. 

2. País o región:

La columna de país o región indica el nombre del país que esta siendo evaluado en el idioma inglés.

3. Puntuación:

La suma de todos los factores siguientes da como resultado el puntaje de felicidad, que es el que se contiene en la columna de puntuación. Son número flotantes que van del 2.853 al 7.769.

4. PIB per cápita, Apoyo Social, Esperanza de vida, Libertad, Generocidad, Percepcion de corrupción:

Para explicar estas columnas primero es importante definir la distopía. 

La distopía es un país imaginario que tiene a las personas menos felices del mundo. El propósito de establecer Distopía es tener un punto de referencia con el cual todos los países puedan ser comparados favorablemente (ningún país se desempeña peor que Distopía) en términos de cada una de las seis variables clave, permitiendo así que cada sub-barra tenga un ancho positivo. Las puntuaciones más bajas observadas para las seis variables clave, por lo tanto, caracterizan a la distopía. Dado que la vida sería muy desagradable en un país con los ingresos más bajos del mundo, la menor esperanza de vida, la menor generosidad, la mayor parte de la corrupción, la menor libertad y el menor apoyo social, se lo conoce como "distopía", en contraste con la utopía.

Ahora bien, las columnas: PIB per cápita, Apoyo Social, Esperanza de vida, Libertad, Generosidad y Percepción de Corrupción describen en qué medida estos factores contribuyen a evaluar la felicidad en cada país tomando como punto de comparación los valores otorgados a Distopía. 

### Clasificación y Agrupacion de datos
El objetivo es una clasificación con los datos, asi que hay que acomodar los datos de manera que se nos facilite esta tarea.

In [18]:
df.Puntuacion.unique()

array([7.769, 7.6  , 7.554, 7.494, 7.488, 7.48 , 7.343, 7.307, 7.278,
       7.246, 7.228, 7.167, 7.139, 7.09 , 7.054, 7.021, 6.985, 6.923,
       6.892, 6.852, 6.825, 6.726, 6.595, 6.592, 6.446, 6.444, 6.436,
       6.375, 6.374, 6.354, 6.321, 6.3  , 6.293, 6.262, 6.253, 6.223,
       6.199, 6.198, 6.192, 6.182, 6.174, 6.149, 6.125, 6.118, 6.105,
       6.1  , 6.086, 6.07 , 6.046, 6.028, 6.021, 6.008, 5.94 , 5.895,
       5.893, 5.89 , 5.888, 5.886, 5.86 , 5.809, 5.779, 5.758, 5.743,
       5.718, 5.697, 5.693, 5.653, 5.648, 5.631, 5.603, 5.529, 5.525,
       5.523, 5.467, 5.432, 5.43 , 5.425, 5.386, 5.373, 5.339, 5.323,
       5.287, 5.285, 5.274, 5.265, 5.261, 5.247, 5.211, 5.208, 5.197,
       5.192, 5.191, 5.175, 5.082, 5.044, 5.011, 4.996, 4.944, 4.913,
       4.906, 4.883, 4.812, 4.799, 4.796, 4.722, 4.719, 4.707, 4.7  ,
       4.696, 4.681, 4.668, 4.639, 4.628, 4.587, 4.559, 4.548, 4.534,
       4.519, 4.516, 4.509, 4.49 , 4.466, 4.461, 4.456, 4.437, 4.418,
       4.39 , 4.374,

In [40]:
Mayor_a_7 = df[df['Puntuacion']>7]
Mayor_a_7

Clasificacion_General   Pais_o_region  Puntuacion  PIB_per_capita  \
0                       1         Finland       7.769           1.340   
1                       2         Denmark       7.600           1.383   
2                       3          Norway       7.554           1.488   
3                       4         Iceland       7.494           1.380   
4                       5     Netherlands       7.488           1.396   
5                       6     Switzerland       7.480           1.452   
6                       7          Sweden       7.343           1.387   
7                       8     New Zealand       7.307           1.303   
8                       9          Canada       7.278           1.365   
9                      10         Austria       7.246           1.376   
10                     11       Australia       7.228           1.372   
11                     12      Costa Rica       7.167           1.034   
12                     13          Israel       7.139           1.276   
13                     14      Luxembourg       7.090           1.609   
14                     15  United Kingdom       7.054           1.333   
15                     16         Ireland       7.021           1.499   

    Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0          1.587              0.986     0.596        0.153   
1          1.573              0.996     0.592        0.252   
2          1.582              1.028     0.603        0.271   
3          1.624              1.026     0.591        0.354   
4          1.522              0.999     0.557        0.322   
5          1.526              1.052     0.572        0.263   
6          1.487              1.009     0.574        0.267   
7          1.557              1.026     0.585        0.330   
8          1.505              1.039     0.584        0.285   
9          1.475              1.016     0.532        0.244   
10         1.548              1.036     0.557        0.332   
11         1.441              0.963     0.558        0.144   
12         1.455              1.029     0.371        0.261   
13         1.479              1.012     0.526        0.194   
14         1.538              0.996     0.450        0.348   
15         1.553              0.999     0.516        0.298   

    Percepción_de_corrupcion  
0                      0.393  
1                      0.410  
2                      0.341  
3                      0.118  
4                      0.298  
5                      0.343  
6                      0.373  
7                      0.380  
8                      0.308  
9                      0.226  
10                     0.290  
11                     0.093  
12                     0.082  
13                     0.316  
14                     0.278  
15                     0.310

In [56]:
Mayor_a_7.to_csv('clasificacion1.csv')

### Responder las preguntas de interes.

- ¿Para que me puede servir estas agrupaciones para la práctica siguiente? 

Para tener más orden en la siguiente práctica y que sea mas fácil encontrar y hacer lo que deseemos hacer. Mas que nada organizar los datos para tenerlos agrupados en base a características específicas y así tener más control sobre los datos. Con esta agrupacion en especifico podemos visualizar mas facilmente cuales son aquellos paises que son "mas felices".

## Limpieza de datos
<b> a) Mostrar los datos sin manipulación: </b>

In [21]:
df

Clasificacion_General             Pais_o_region  Puntuacion  \
0                        1                   Finland       7.769   
1                        2                   Denmark       7.600   
2                        3                    Norway       7.554   
3                        4                   Iceland       7.494   
4                        5               Netherlands       7.488   
..                     ...                       ...         ...   
151                    152                    Rwanda       3.334   
152                    153                  Tanzania       3.231   
153                    154               Afghanistan       3.203   
154                    155  Central African Republic       3.083   
155                    156               South Sudan       2.853   

     PIB_per_capita  Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0             1.340         1.587              0.986     0.596        0.153   
1             1.383         1.573              0.996     0.592        0.252   
2             1.488         1.582              1.028     0.603        0.271   
3             1.380         1.624              1.026     0.591        0.354   
4             1.396         1.522              0.999     0.557        0.322   
..              ...           ...                ...       ...          ...   
151           0.359         0.711              0.614     0.555        0.217   
152           0.476         0.885              0.499     0.417        0.276   
153           0.350         0.517              0.361     0.000        0.158   
154           0.026         0.000              0.105     0.225        0.235   
155           0.306         0.575              0.295     0.010        0.202   

     Percepción_de_corrupcion  
0                       0.393  
1                       0.410  
2                       0.341  
3                       0.118  
4                       0.298  
..                        ...  
151                     0.411  
152                     0.147  
153                     0.025  
154                     0.035  
155                     0.091  

[156 rows x 9 columns]

<b> b) Eliminar las columnas no deseadas: </b>

>En nuestro caso, como el objetivo de nuestro trabajo es analizar cuales son aquellos factores que influyen en la felicidad de las personas, hemos decidido que no es necesario eliminar ninguna columna. Esto ya que cada columna representa los diversos factores que analizaremos

<b> c) Correcta manipulación de los datos nulos dentro de la base de datos: </b>

In [22]:
df.isnull()

Clasificacion_General  Pais_o_region  Puntuacion  PIB_per_capita  \
0                    False          False       False           False   
1                    False          False       False           False   
2                    False          False       False           False   
3                    False          False       False           False   
4                    False          False       False           False   
..                     ...            ...         ...             ...   
151                  False          False       False           False   
152                  False          False       False           False   
153                  False          False       False           False   
154                  False          False       False           False   
155                  False          False       False           False   

     Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0           False              False     False        False   
1           False              False     False        False   
2           False              False     False        False   
3           False              False     False        False   
4           False              False     False        False   
..            ...                ...       ...          ...   
151         False              False     False        False   
152         False              False     False        False   
153         False              False     False        False   
154         False              False     False        False   
155         False              False     False        False   

     Percepción_de_corrupcion  
0                       False  
1                       False  
2                       False  
3                       False  
4                       False  
..                        ...  
151                     False  
152                     False  
153                     False  
154                     False  
155                     False  

[156 rows x 9 columns]

>Aqui podemos Observar que no existen datos nulos, pero para comprobarlo usarlos la funcion isnull().any()

In [23]:
df.isnull().any()

Clasificacion_General       False
Pais_o_region               False
Puntuacion                  False
PIB_per_capita              False
Apoyo_Social                False
Esperanza_de_vida           False
Libertad                    False
Generocidad                 False
Percepción_de_corrupcion    False
dtype: bool

>Con esto, confirmamos que no existen datos nulos en nuestra tabla.

<b> e) Crear un nuevo archivo con la base de datos editada

In [28]:
df.to_csv('newdata.csv')

> Y asi, tenemos nuestra base de datos editada en csv

---
## Base de datos 2

### Titulo de la base de datos.

<b> a) Nombre con el que se encuentra en la página encontrada: </b> Disney Movies <br>
<b> b) Url de la página: </b> https://www.kaggle.com/prateekmaj21/disney-movies

### Preparación de datos
1. Importar librerías

In [4]:
import pandas as pd
import numpy as np
import json

2. Importar base de datos:

In [26]:
datos = pd.read_csv("uncleandata_disney.csv")

### Visualización de los Datos


In [21]:
datos

movie_title release_date      genre mpaa_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

     total_gross  inflation_adjusted_gross  
0      184925485                5228953251  
1       84300000                2188229052  
2       83320000                2187090808  
3       65000000                1078510579  
4       85000000                 920608730  
..           ...                       ...  
574     12545979                  12545979  
575      8874389                   8874389  
576    232532923                 232532923  
577    246082029                 246082029  
578    529483936                 529483936  

[579 rows x 6 columns]

1. Revisar cuales son los datos que conforman la base de datos:

In [6]:
datos.shape
print("Numero de filas:", datos.shape[0], "\nNumero de columnas:", datos.shape[1])

Numero de filas: 579 
Numero de columnas: 6


2. Ver los primeros cinco elementos de cada una de las columnas

In [7]:
datos.head()

movie_title release_date      genre mpaa_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  
0    184925485                5228953251  
1     84300000                2188229052  
2     83320000                2187090808  
3     65000000                1078510579  
4     85000000                 920608730

3. Ver los últimos cinco elementos de cada una de las columnas

In [8]:
datos.tail()

movie_title release_date      genre mpaa_rating  \
574      The Light Between Oceans   2016-09-02      Drama       PG-13   
575                Queen of Katwe   2016-09-23      Drama          PG   
576                Doctor Strange   2016-11-04  Adventure       PG-13   
577                         Moana   2016-11-23  Adventure          PG   
578  Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

     total_gross  inflation_adjusted_gross  
574     12545979                  12545979  
575      8874389                   8874389  
576    232532923                 232532923  
577    246082029                 246082029  
578    529483936                 529483936

4. Ver los nombres de las columnas de nuestra base de datos

In [9]:
datos.columns

Index(['movie_title', 'release_date', 'genre', 'mpaa_rating', 'total_gross',
       'inflation_adjusted_gross'],
      dtype='object')

5. Cambiar los nombres de las columnas a uno que nos facilite la extracción de su información

In [28]:
datos = datos.rename(columns={'movie_title': 'Titulo_Pelicula', 
                              'release_date':'Fecha_Estreno',
                              'genre':'Genero',
                              'mpaa_rating': 'Clasificacion',
                              'total_gross':'Ganancias',
                              'inflation_adjusted_gross':'Ganancias_con_Inflacion'})

datos

Titulo_Pelicula Fecha_Estreno     Genero Clasificacion  \
0    Snow White and the Seven Dwarfs    1937-12-21    Musical             G   
1                          Pinocchio    1940-02-09  Adventure             G   
2                           Fantasia    1940-11-13    Musical             G   
3                  Song of the South    1946-11-12  Adventure             G   
4                         Cinderella    1950-02-15      Drama             G   
..                               ...           ...        ...           ...   
574         The Light Between Oceans    2016-09-02      Drama         PG-13   
575                   Queen of Katwe    2016-09-23      Drama            PG   
576                   Doctor Strange    2016-11-04  Adventure         PG-13   
577                            Moana    2016-11-23  Adventure            PG   
578     Rogue One: A Star Wars Story    2016-12-16  Adventure         PG-13   

     Ganancias  Ganancias_con_Inflacion  
0    184925485               5228953251  
1     84300000               2188229052  
2     83320000               2187090808  
3     65000000               1078510579  
4     85000000                920608730  
..         ...                      ...  
574   12545979                 12545979  
575    8874389                  8874389  
576  232532923                232532923  
577  246082029                246082029  
578  529483936                529483936  

[579 rows x 6 columns]

6. Conocer el tipo de dato que contiene cada columna

In [29]:
datos.dtypes

Titulo_Pelicula            object
Fecha_Estreno              object
Genero                     object
Clasificacion              object
Ganancias                   int64
Ganancias_con_Inflacion     int64
dtype: object

- Si queremos conocer el tipo de datos de una columna en específico:

In [30]:
datos['Genero'].dtype

dtype('O')

### Descripción Básica de las columnas

In [31]:
datos

Titulo_Pelicula Fecha_Estreno     Genero Clasificacion  \
0    Snow White and the Seven Dwarfs    1937-12-21    Musical             G   
1                          Pinocchio    1940-02-09  Adventure             G   
2                           Fantasia    1940-11-13    Musical             G   
3                  Song of the South    1946-11-12  Adventure             G   
4                         Cinderella    1950-02-15      Drama             G   
..                               ...           ...        ...           ...   
574         The Light Between Oceans    2016-09-02      Drama         PG-13   
575                   Queen of Katwe    2016-09-23      Drama            PG   
576                   Doctor Strange    2016-11-04  Adventure         PG-13   
577                            Moana    2016-11-23  Adventure            PG   
578     Rogue One: A Star Wars Story    2016-12-16  Adventure         PG-13   

     Ganancias  Ganancias_con_Inflacion  
0    184925485               5228953251  
1     84300000               2188229052  
2     83320000               2187090808  
3     65000000               1078510579  
4     85000000                920608730  
..         ...                      ...  
574   12545979                 12545979  
575    8874389                  8874389  
576  232532923                232532923  
577  246082029                246082029  
578  529483936                529483936  

[579 rows x 6 columns]

<b>Descripción: </b> 

1. Titulo Película: La columna de título de película indica el nombre de la película que está siendo evaluada
2. Fecha Estreno: La columna de Fecha de Estreno indica la fecha en que tuvo su estreno la película que esta siendo evaluada
3. Genero: La columna de Género indica el género de la película que está siendo evaluada
4. Clasificacion: La columna de clasificacion nos indica la clasificacion de la pelicula que esta siendo evaluada en base al Sistema de calificaciones de la Motion Picture Association (Motion Picture Association film rating system) 
5. Ganancias: La columna de Ganancias indica las ganancias que tuvo la película que está siendo evaluada
6. Ganancias con inflacion: La columna de Ganancias con inflación da los valores de las ganancias ya con la inflación de la película evaluada

### Clasificación de datos
El objetivo es una clasificación con los datos, asi que hay que acomodar los datos de manera que se nos facilite esta tarea.

In [32]:
datos.Genero.unique()

array(['Musical', 'Adventure', 'Drama', 'Comedy', nan, 'Action', 'Horror',
       'Romantic Comedy', 'Thriller/Suspense', 'Western', 'Black Comedy',
       'Documentary', 'Concert/Performance'], dtype=object)

In [16]:
Genero_musical= datos[datos['Genero']=='Musical']
Genero_musical

Titulo_Pelicula Fecha_Estreno   Genero  \
0           Snow White and the Seven Dwarfs    1937-12-21  Musical   
2                                  Fantasia    1940-11-13  Musical   
10                         Babes in Toyland    1961-12-14  Musical   
13                          The Jungle Book    1967-10-18  Musical   
15                           The Aristocats    1970-04-24  Musical   
17                 Bedknobs and Broomsticks    1971-10-07  Musical   
114                    Beauty and the Beast    1991-11-13  Musical   
142                              Swing Kids    1993-03-05  Musical   
161          The Nightmare Before Christmas    1993-10-13  Musical   
254                                   Evita    1996-12-25  Musical   
321                    Fantasia 2000 (IMAX)    2000-01-01  Musical   
330      Fantasia 2000 (Theatrical Release)    2000-06-16  Musical   
354             Beauty and the Beast (IMAX)    2002-01-01  Musical   
446  Tim Burton's The Nightmare Before Chr…    2006-10-20  Musical   
474      High School Musical 3: Senior Year    2008-10-24  Musical   
553                          Into the Woods    2014-12-25  Musical   

    Clasificacion  Ganancias  Ganancias_con_Inflacion  
0               G  184925485               5228953251  
2               G   83320000               2187090808  
10              G   10218316                124841160  
13      Not Rated  141843000                789612346  
15              G   55675257                255161499  
17            NaN   17871174                 91305448  
114             G  218951625                363017667  
142         PG-13    5632086                 11468231  
161            PG   50408318                100026637  
254            PG   50047179                 92077628  
321             G   60507228                 94852354  
330             G    9103630                 14238144  
354             G   25487028                 36980311  
446            PG   24732041                 30737517  
474             G   90559416                106308538  
553            PG  128002372                130894237

In [17]:
Genero_musical.to_csv('Genero_Musical.csv')

### Responder preguntas de interés

- ¿Para que me puede servir estas agrupaciones para la práctica siguiente? 

Para visualizar los datos de solo aquellas peliculas que sean de un genero en especifico, en este caso del genero musical

## Limpieza de datos
<b> a) Mostrar los datos sin manipulación: </b>

In [33]:
datos

Titulo_Pelicula Fecha_Estreno     Genero Clasificacion  \
0    Snow White and the Seven Dwarfs    1937-12-21    Musical             G   
1                          Pinocchio    1940-02-09  Adventure             G   
2                           Fantasia    1940-11-13    Musical             G   
3                  Song of the South    1946-11-12  Adventure             G   
4                         Cinderella    1950-02-15      Drama             G   
..                               ...           ...        ...           ...   
574         The Light Between Oceans    2016-09-02      Drama         PG-13   
575                   Queen of Katwe    2016-09-23      Drama            PG   
576                   Doctor Strange    2016-11-04  Adventure         PG-13   
577                            Moana    2016-11-23  Adventure            PG   
578     Rogue One: A Star Wars Story    2016-12-16  Adventure         PG-13   

     Ganancias  Ganancias_con_Inflacion  
0    184925485               5228953251  
1     84300000               2188229052  
2     83320000               2187090808  
3     65000000               1078510579  
4     85000000                920608730  
..         ...                      ...  
574   12545979                 12545979  
575    8874389                  8874389  
576  232532923                232532923  
577  246082029                246082029  
578  529483936                529483936  

[579 rows x 6 columns]

<b> b) Eliminar las columnas no deseadas: </b>

Las columnas que no necesitaremos son: Titulo_Pelicula, Fecha_Estreno y Ganancias_con_Inflacion

In [34]:
datos.drop(['Titulo_Pelicula', 'Fecha_Estreno', 'Ganancias_con_Inflacion'], axis=1)

Genero Clasificacion  Ganancias
0      Musical             G  184925485
1    Adventure             G   84300000
2      Musical             G   83320000
3    Adventure             G   65000000
4        Drama             G   85000000
..         ...           ...        ...
574      Drama         PG-13   12545979
575      Drama            PG    8874389
576  Adventure         PG-13  232532923
577  Adventure            PG  246082029
578  Adventure         PG-13  529483936

[579 rows x 3 columns]

<b> c) Correcta manipulación de los datos nulos dentro de la base de datos: </b>

In [41]:
datos

Genero Clasificacion  Ganancias
0      Musical             G  184925485
1    Adventure             G   84300000
2      Musical             G   83320000
3    Adventure             G   65000000
4        Drama             G   85000000
..         ...           ...        ...
574      Drama         PG-13   12545979
575      Drama            PG    8874389
576  Adventure         PG-13  232532923
577  Adventure            PG  246082029
578  Adventure         PG-13  529483936

[579 rows x 3 columns]

>Ahora comprobaremos si no existen datos nulos

In [42]:
datos.isnull()

Genero  Clasificacion  Ganancias
0     False          False      False
1     False          False      False
2     False          False      False
3     False          False      False
4     False          False      False
..      ...            ...        ...
574   False          False      False
575   False          False      False
576   False          False      False
577   False          False      False
578   False          False      False

[579 rows x 3 columns]

>A primera vista parace que no, pero usaremos la funcion isnull().any() para comprobarlo

In [43]:
datos.isnull().any()

Genero            True
Clasificacion     True
Ganancias        False
dtype: bool

>Al parecer si existen datos nulos, debido a que estan en nuestras categorias de genero y clasificacion, los pondremos como "Desconocido".

In [46]:
datos.fillna('Desconocido',inplace=True)

<b> e) Crear un nuevo archivo con la base de datos editada

In [48]:
datos.to_csv('newdata_disney.csv')